src/core/thread/mle_router.cpp

```diff
@@ -2112,7 +2111,7 @@ otError MleRouter::UpdateChildAddresses(const AddressRegistrationTlv &aTlv, Chil
             address = *entry->GetIp6Address();
         }
 
-        aChild.GetIp6Address(count) = address;
+        SuccessOrExit(error = aChild.AddIp6Address(address));
```

Comment:
We don't actually want to fail here.  We should try to accept as many addresses as we can and then provide a Child ID/Update Response indicating what addresses were accepted.

Reply:
Good point! Will change it.